iml-manager-cli/src/server.rs

```diff
@@ -61,17 +73,61 @@ pub enum ServerCommand {
     Remove(RemoveHosts),
 }
 
-#[derive(Debug, serde::Serialize, serde::Deserialize)]
+#[derive(Debug)]
+pub enum AuthType {
+    Shared,
+    Password(String),
+    PrivateKey(String),
+}
+
+impl From<&AuthType> for String {
+    fn from(auth: &AuthType) -> Self {
+        match auth {
+            AuthType::Shared => "existing_keys_choice".into(),
+            AuthType::Password(_) => "id_password_root".into(),
+            AuthType::PrivateKey(_) => "private_key_choice".into(),
+        }
+    }
+}
+
+#[derive(Debug, serde::Serialize, serde::Deserialize, Default)]
 struct TestHostConfig<'a> {
     address: &'a str,
     auth_type: String,
+    // For auth_type == "id_password_root"
+    root_password: Option<String>,
+    // For auth_type == "private_key_choice"
+    private_key: Option<String>,
+}
+
+impl<'a> TestHostConfig<'a> {
+    fn new(address: &'a str, auth: &AuthType) -> Self {
+        Self {
+            address,
+            auth_type: auth.into(),
+            root_password: {
```

Comment:
Not going to hold up the PR but I think we can simplify this by doing the following:

```
impl From<&AuthType> for Option<String> {
    fn from(auth: &AuthType) -> Self {
        match auth {
            AuthType::Password(p) => Some(p.into()),
            AuthType::PrivateKey(p) => Some(p.into()),
            _ => None
        }
    }
}

impl<'a> TestHostConfig<'a> {
    fn new(address: &'a str, auth: &AuthType) -> Self {
        Self {
            address,
            auth_type: auth.into(),
            root_password: auth.into(),
            private_key: auth.into()
        }
    }
}
```
Let me know your thoughts.

Reply:
Seems reasonable, I think this change highlights how the API struct is not really designed correctly.

I.E. since root_password and private key are disjoint, it really doesn't make sense to have two fields there. 

They should probably be collapsed into a single field which could hold either one of them.